packages/lumx-react/src/components/text/Text.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { COMPONENT_PREFIX } from '@lumx/react/constants';
+import { EllipsisType, GenericProps, truncate } from '@lumx/react/utils';
+
+/**
+ * Defines the props of the component.
+ */
+interface TextProps extends GenericProps {
+    /** The text to display */
+    text: string;
+    /** The max number of characteres to display */
+    maxChars?: number;
+    /** Works with maxChars. The ellipsis method to use */
+    ellipsisType?: EllipsisType;
+}
+
+/**
+ * The display name of the component.
+ */
+const COMPONENT_NAME = `${COMPONENT_PREFIX}Text`;
+
+/**
+ * The default value of props.
+ */
+const DEFAULT_PROPS: Partial<TextProps> = {
+    maxChars: -1,
+    ellipsisType: EllipsisType.END,
+};
+
+const Text: React.FC<TextProps> = ({ text, ellipsisType, maxChars, ...forwardedProps }) => {
+    // const shouldTruncate = ellipsisType && maxChars && maxChars > 0 && text.length > maxChars;
+    const a11yProps: any = {};
+    const resultingText = truncate(text, { max: maxChars as number, ellipsisType });
+
+    if (resultingText !== text) {
+        a11yProps.ariaLabel = text;
```

Comment:
I wouldn't put the `aria-label` here. At least not automatically.

I think this is something that should be handled by the consumer of the component. Here you are saying that sighted users should see a truncated version of the text but visually impaired users should have the entire text. In some cases, this wouldn't be applicable, for example if you are truncated a huge amount of text.

If you look at the search page result of W3 site: https://www.w3.org/WAI/search/?q=accessible
You'll see multiple truncated result without any particular aria-label. Sighted and visually impaired users would have the same experience.

Concerning small texts like title and link, it is common to truncate them but still offer to the user the access to the complete information. In this case, there will be an aria-label for visually impaired users but for sighted users, you would have a tooltip.
This makes me think that the `aria-label` shouldn't be set by the `Text` component. It should be handled by the developer when using `Text` by surrounding it by a `Tooltip` which handles the `aria-label` itself.

What do you think?